pkg/kopia/command/storage/secret_utils.go

```diff
@@ -56,3 +68,123 @@ func checkSkipSSLVerifyFromMap(m map[string]string) bool {
 func locationType(m map[string]string) LocType {
 	return LocType(m[typeKey])
 }
+
+// GenerateEnvSpecFromCredentialSecret parses the secret and returns
+// list of EnvVar based on secret type
+func GenerateEnvSpecFromCredentialSecret(s *v1.Secret) ([]v1.EnvVar, error) {
+	if s == nil {
+		return nil, errors.New("Secret cannot be nil")
+	}
+	secType := string(s.Type)
+	switch secType {
+	case secrets.AWSSecretType:
+		return getEnvSpecForAWSCredentialSecret(s)
+	case secrets.AzureSecretType:
+		return getEnvSpecForAzureCredentialSecret(s)
+	}
+	// We only need to set the environment variables in cases where
+	// secret type is AWS or Azure.
+	return nil, nil
+}
+
+func getEnvSpecForAWSCredentialSecret(s *v1.Secret) ([]v1.EnvVar, error) {
+	var duration time.Duration
+	var err error
+	envVars := []v1.EnvVar{}
+	envVars = append(
+		envVars,
+		getEnvVarWithSecretRef(aws.AccessKeyID, s.Name, secrets.AWSAccessKeyID),
+		getEnvVarWithSecretRef(aws.SecretAccessKey, s.Name, secrets.AWSSecretAccessKey),
+	)
+	if val, ok := s.Data["assume_duration"]; ok {
+		duration, err = time.ParseDuration(string(val))
+		if err != nil {
+			return nil, errors.Wrap(err, "Failed to parse AWS Assume Role duration")
+		}
+	}
+	creds, err := secrets.ExtractAWSCredentials(context.Background(), s, duration)
+	if err != nil {
+		return nil, err
+	}
+	if creds.SessionToken != "" {
+		envVars = append(envVars, getEnvVarWithSecretRef(aws.SessionToken, s.Name, secrets.AWSSessionToken))
+	}
+	return envVars, nil
+}
+
+func getEnvSpecForAzureCredentialSecret(s *v1.Secret) ([]v1.EnvVar, error) {
+	envVars := []v1.EnvVar{}
+	envVars = append(
+		envVars,
+		getEnvVarWithSecretRef("AZURE_STORAGE_ACCOUNT", s.Name, secrets.AzureStorageAccountID),
```

Comment:
can we declare "AZURE_STORAGE_ACCOUNT", "AZURE_STORAGE_KEY",  "AZURE_STORAGE_DOMAIN" as constant?